src/Microsoft.ML.Data/Training/EarlyStoppingCriteria.cs

```diff
@@ -21,7 +21,7 @@
 [assembly: EntryPointModule(typeof(PQEarlyStoppingCriterion))]
 [assembly: EntryPointModule(typeof(UPEarlyStoppingCriterion))]
 
-namespace Microsoft.ML.Internal.Internallearn
+namespace Microsoft.ML.Trainers.FastTree
```

Comment:
>namespace Microsoft.ML.Trainers [](start = 0, length = 31)

would this be better under just `Microsoft.ML.Trainers`, since more than FastTree might use it. Does LightGBM uses its?

Reply:
I believe it’s just FastTree

Reply:
So one thing that I am not seeing yet, is I do not see that this has moved to FastTree? Is that coming in a future commit? If we're saying something is useful only to one assembly even to the point where we are changing its namespace, why not just move that to that assembly outright?

---
In reply to: [260433610](https://github.com/dotnet/machinelearning/pull/2740#discussion_r260433610) [](ancestors = 260433610)

Reply:
Also double check that this is only being used in FastTree also on internal code. That may change our thinking here, just something to double check.

---
In reply to: [260449769](https://github.com/dotnet/machinelearning/pull/2740#discussion_r260449769) [](ancestors = 260449769,260433610)

Reply:
Seems to also being used in NeuralNetworks but as we discussed offline we probably don't care for now. I have moved this file in fast tree project.